File: backend/IoTDeviceAPI/Models/IoTDevice.cs
namespace IoTDeviceAPI.Models
{
    public class IoTDevice
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public bool IsOn { get; set; }
        public DeviceType DeviceType { get; set; }
    }
}

File: frontend/src/types/IoTDevice.ts
export enum DeviceType {
  Lightbulb = 0,
  AirConditioner = 1
}

export interface IoTDevice {
  id: number;
  name: string;
  isOn: boolean;
  deviceType: DeviceType;
}

export interface BatchToggleRequest {
  deviceType: DeviceType;
  state: boolean;
}

export interface BatchToggleResult {
  success: boolean;
  devicesAffected: number;
  totalDevices: number;
  message: string;
  errors: string[];
}